### PR TITLE
fix: prevent infinite GetTwinOperation loop

### PR DESCRIPTION
**Pull request by lev-i**
- Added connection logic to on_get_twin_complete to ensure GetTwinOperations aren't tried indefinitely when not connected
- Added appropriate unit tests
- Fixed minor typos

### CODE REVIEW

File: azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py [hunk 1]
```diff
@@ -668,6 +683,86 @@ def test_does_not_change_last_version_seen_if_same_version(
...
+        mocker.spy(stage, "report_background_exception")
+        return stage
+
+    @pytest.fixture
+    def get_twin_op(self, stage):
+        stage.last_version_seen = -1
+        stage.handle_pipeline_event(pipeline_events_base.ConnectedEvent())
+
+        get_twin_op = stage.send_op_down.call_args[0][0]
+        assert isinstance(get_twin_op, pipeline_ops_iothub.GetTwinOperation)
```
Comment (BertKleewein) [inline azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py:708]: Yes, perfect. For extra paranoia, I would assert the value of  `send_op_down.call_count` before and after the `ConnectedEvent` to make sure we're working with the `GetTwinOperation` that came from the `ConnectedEvent`

**Review by BertKleewein — APPROVED** [review]
:shipit:

File: azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py (hunk 1 above)
Comment (lev-i) [reply azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub.py:709]: Good suggestion, thanks!

**Pull request merged**